### DISCUSSION TITLE
Treat Recipes as Foods dynamically

### DISCUSSION THREAD

**Discussion by PeterAttardo** [TandoorRecipes/recipes#3301] [Ideas]
Currently when you create a recipe, you can check the `create food` option to create a food entry so that you can include that recipe as an ingredient in other recipes. This appears to be a static creation however; simply creating a new entry in the food table and auto-populating some values. It would be nice if they were tied together in a more dynamic way, where the resulting food carries over the properties of the recipe, and updates those values if and when the recipe is altered. Because all recipes are quantized in servings, the `Properties Food Amount` and `Properties Food Unit` could default to `1 Serving`, and conversions from `Serving` to weight/volumetric measurements could be added on a per-recipe basis.

**Comment by vabene1111** [comment]
that would actually be pretty cool. Sadly also a bit complicated, I am already struggeling a bit with the dependency tree after updating a food/recipe with properties. 

What I definitely need to do is include sub recipes in the property calculation, there I could also include calculation for ingredients that are recipes, that would solve this issue as well. Maybe I should also disable putting properties on foods that have a recipe linked. 

All of these are topics for after #2976

**Reply by smilerz** [reply]
This is a good use case for either batch calculations (via celery) or materialized views.  Things like food properties related to substitution/on hand should probably be handled that way too.

**Reply by vabene1111** [reply]
likely, the problem with the properties is the completely undefined nature of the topic because both rows and columns are completely arbitrary (1 to n foods, 1 to n properties) and those relations can also bee infinite levels deep. 

One will need to pay very close attention on how such a system is implemented
